### PR TITLE
Enhancement for simplify_expr

### PR DESCRIPTION
**Pull request by leiwen83**
* Add SwitchAdd0 to switch constant+var to var+constant to make fusion easy

* Add SwitchAdd1 to switch var+var+constant to var+constant+var to make fusion easy

* Add SimplifyDivide to prepare it for fold_scale_axis pass

Signed-off-by: Lei Wen <wenlei03@qiyi.com>

### CODE REVIEW

File: src/relay/transforms/simplify_expr.cc [hunk 1]
```diff
@@ -878,6 +932,33 @@ class SimplifyAdd : public DFPatternRewrite {
...
+
+  Expr Callback(const Expr& pre, const Expr& post,
+                const Map<DFPattern, Array<Expr>>& node_map) const override {
+    Type pre_type = pre->checked_type_;
+    auto dtype = pre_type.as<TensorTypeNode>()->dtype;
+    auto x = node_map[x_][0];
+    auto c = node_map[c_][0];
+
+    Expr one;
+    one = MakeConstantScalar(dtype, 1);
```
Comment (ehsanmok) [inline src/relay/transforms/simplify_expr.cc:952]: Minor: var declare and initialize at the same time.

**Review by junrushao — APPROVED** [review]
LGTM

File: src/relay/transforms/simplify_expr.cc (hunk 1 above)
Comment (leiwen83) [reply src/relay/transforms/simplify_expr.cc]: fixed

**Comment by junrushao** [comment]
i'm retriggering this pr

**Comment by junrushao** [comment]
Please fix the unit tests

**Comment by leiwen83** [comment]
> 

There are two passes fail the checkpatch, and I don't know how to fix it ... 
So I remove them at current time, and now patch pass all unit test. @junrushao

**Comment by junrushao** [comment]
Unfortunately I may not have enough bandwidth lately, but am happy to help if more details are provided on your end :-)  Specifically, are you able to reproduce locally the issues you encountered in CI?

**Comment by leiwen83** [comment]
> Unfortunately I may not have enough bandwidth lately, but am happy to help if more details are provided on your end :-) Specifically, are you able to reproduce locally the issues you encountered in CI?

Sorry for the late response. I haven't reproduce the previous issue in the CI.@junrushao 
Now the patch is shortten to pass the CI, so I wander that how about merge the current status? And I would check the CI issue later when I find time for the original PR  patch.

**Comment by leiwen83** [comment]
BTW, some failure in the CI is wield..
Like https://ci.tlcpack.ai/blue/organizations/jenkins/tvm/detail/PR-13294/28/pipeline/
All tests in https://ci.tlcpack.ai/blue/organizations/jenkins/tvm/detail/PR-13294/28/tests has passed, but CI still failed, and I cannot get any clue which case it fail at...

**Pull request closed without merging**